lxd/instances_post.go

```diff
@@ -1028,77 +1044,84 @@ func instancesPost(d *Daemon, r *http.Request) response.Response {
 		return response.BadRequest(err)
 	}
 
-	if clustered && (targetNode == "" || targetGroup != "") {
-		// If no target node was specified, pick the node with the least number of instances.
-		// If there's just one node, or if the selected node is the local one, this is effectively a no-op,
-		// since GetNodeWithLeastInstances() will return an empty string.
-		// If the target is a cluster group, find a suitable node within the group.
-
-		// Load restricted groups from project.
-		var allowedGroups []string
-
-		if !isClusterNotification(r) && shared.IsTrue(targetProject.Config["restricted"]) {
-			allowedGroups = shared.SplitNTrimSpace(targetProject.Config["restricted.cluster.groups"], ",", -1, true)
-
-			// Validate restrictions.
-			if targetGroup != "" && !shared.StringInSlice(targetGroup, allowedGroups) {
-				return response.Forbidden(fmt.Errorf("Project isn't allowed to use this cluster group"))
-			}
-		}
-
+	if clustered && !clusterNotification && targetMember == "" {
 		architectures, err := instance.SuitableArchitectures(r.Context(), s, targetProjectName, sourceInst, sourceImageRef, req)
 		if err != nil {
 			return response.BadRequest(err)
 		}
 
-		err = d.db.Cluster.Transaction(r.Context(), func(ctx context.Context, tx *db.ClusterTx) error {
-			defaultArch := ""
-			if targetProject.Config["images.default_architecture"] != "" {
-				defaultArch = targetProject.Config["images.default_architecture"]
-			} else {
+		// If no architectures have been ascertained from the source then use the default architecture
```

Comment:
@stgraber I found the old logic for when to use the default architecture in `GetNodeWithLeastInstances` difficult to follow (I never did quite figure it out - see `isDefaultArch` and `isDefaultArchChosen` and the like).

However I wanted to check that my simpler interpretation of that feature is indeed correct.

That basically if a suitable list of architectures not returned from `instance.SuitableArchitectures` based on the source, then if there is a default architecture specified then we use that instead, and if its not specified then we don't filter cluster member candidates by architecture at all.